pkg/apis/sessions/session_state.go

```diff
@@ -64,6 +64,32 @@ func (s *SessionState) String() string {
 	return o + "}"
 }
 
+func (s *SessionState) GetClaim(claim string) string {
```

Comment:
I consider these more `fields` than `claims` -- If we intend to support generic fields from a session beyond fields we generate from claims (which is likely a good idea to expose those details to users) -- I think this logic should be split from the other logic we need to add of extracting generic claims into the session (and allowing those as well to be added as headers).

Reply:
I'm not following this statement sorry
> I consider these more fields than claims -- If we intend to support generic fields from a session beyond fields we generate from claims (which is likely a good idea to expose those details to users)

Do you think this method should be `GetField`? I realise `access_token` and `id_token` are not really claims (as with some others) so we are overloading the name here a bit.

> I think this logic should be split from the other logic we need to add of extracting generic claims into the session (and allowing those as well to be added as headers).

Are you saying we should have two separate fields for the header injection? One for extra claims and one for generic fields in the session? This bit has got me confused 😅

Reply:
So the main area I had my eye on was generic Claim -> Session -> Headers mapping.  IE allowing users to extract extra claims like `phone_number`, `full_name` etc, put those into the session appropriately, and then for `X-Forwarded-*` & `X-Auth-Request-*` headers.

So a user can say `phone_number` claim maps to `Phone-Number` and requests/resposes have the `X-Forwarded-Phone-Number` & `X-Auth-Request-Phone-Number` with that claim's value.

And add these to the session appropriately so we can support future more flexible Authorization rules on these potentially as well.